### ISSUE TITLE
Astropy units with fractional dimensions are buggy 

### ISSUE THREAD

**Issue by astroduff** [astropy/astropy#9279]
Hey all, as raised in issue https://github.com/astropy/astropy/issues/6537#issue-256938682 for astropy 2.x, the latest astropy 3.2.1 still has a bug with units where fractions are incorrectly simplified 
```python
In [1]: from astropy import units as u                                          
In [2]: u.kg**(7/11)                                                            
Out[2]: Unit("kg(5/8)")
```
This needs to raise an error as 7/11 is not 5/8 and the user should be alerted to this bug, or even better implement into units the previous 'fix' of using external package 'fractions' suggested here https://github.com/astropy/astropy/issues/6537#issuecomment-328858001

EDIT: Code formatting.

**Comment by mhvk** [comment]
Yikes, that is awful!  I'm very sorry I missed your original report. I had some vague hope my recent effort to get powers be better behaved (#9267) would have helped, but no such luck.

One fortunate thing is that it actually does not affect the *actual* power:
```
t = u.kg**(7/11)
t.powers
# [0.6363636363636364]
t ** 11
# Unit("kg7")
```

**Comment by mhvk** [comment]
Investigating a bit more, it looks like there is a bug in how the powers are represented in `__repr__` and `__str__` (and, therefore, in `format.Generic.to_string()`, which goes on to the real culprit:
```
from astropy.units.format.utils import format_power
format_power(7/11)
# '5/8'
```
I'll try to create a fix...

**Comment by 1313e** [comment]
When I raised this issue two years ago in #6537, I mentioned that whenever there is a fraction (in the power of a unit) whose denominator cannot be written as `10` or smaller, it is approximated to a fraction that can.
Maybe that will give an idea of why this behavior exists.

**Comment by mhvk** [comment]
@1313e - yes, indeed that is where the problem is. I now added a fix to #9267.
